qiskit/synthesis/su4/siswap_decompose.py

```diff
@@ -0,0 +1,286 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name, non-ascii-name
+
+"""Synthesis of two-qubit unitaries using at most 3 applications of the sqrt(iSWAP) gate."""
+
+import cmath
+from typing import Optional, List, Union
+import numpy as np
+
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+from qiskit.quantum_info.operators import Operator
+from qiskit.quantum_info.synthesis.two_qubit_decompose import TwoQubitWeylDecomposition
+from qiskit.circuit.library import SiSwapGate, XGate, YGate, ZGate
+from qiskit.synthesis.su4.utils import find_min_point, average_infidelity
+
+
+_EPS = 1e-12
+
+# the polytope accessible by two applications of the SiSwapGate
+# (red region in paper)
+_ID = np.array([0, 0, 0])
+_CNOT = np.array([np.pi / 4, 0, 0])
+_ISWAP = np.array([np.pi / 4, np.pi / 4, 0])
+_MID = np.array([np.pi / 4, np.pi / 8, np.pi / 8])
+_MIDDG = np.array([np.pi / 4, np.pi / 8, -np.pi / 8])
+_SISWAP = np.array([np.pi / 8, np.pi / 8, 0])
+_POLYTOPE = np.array([_ID, _CNOT, _ISWAP, _MID, _MIDDG])
+
+
+class SiSwapDecomposer:
+    """
+    A class for decomposing 2-qubit unitaries into at most 3 uses of the sqrt(iSWAP) gate.
+
+    This basis is attractive in that it is shorter/cheaper than iSWAP and generates a large
+    volume (~79%) of Haar-random SU(4) unitaries with only two applications, and an even
+    larger volume when approximation is enabled.
+
+    Args:
+        euler_basis (list(str)): single-qubit gates basis in the decomposition.
+
+    Reference:
+        1. C. Huang et al,
+           *Towards ultra-high fidelity quantum operations:
+           SQiSW gate as a native two-qubit gate (2021)*
```

Comment:
A bit nitpicky, but I don't think we have slanted fonts anywhere else 🙂 
```suggestion
           Towards ultra-high fidelity quantum operations:
           SQiSW gate as a native two-qubit gate (2021)
```